pysparkling/sql/expressions/mappers.py

```diff
@@ -49,14 +51,24 @@ def __init__(self, item, field):
         self.field = field
 
     def eval(self, row, schema):
-        item_value = self.item.eval(row, schema)
+        try:
+            idx = schema.names.index(self.item.col_name)
+            if isinstance(schema.fields[idx].dataType, StructType):
+                item_value = dict(list(zip(schema.fields[idx].dataType.names, self.item.eval(row, schema))))
+            elif isinstance(schema.fields[idx].dataType, MapType):
+                item_value = self.item.eval(row, schema)
+            else:
+                item_value = dict(enumerate(self.item.eval(row, schema)))
+        except ValueError:
```

Comment:
Do you have in mind a case where this ValueError is thrown?

Reply:
I got the error when running tests on the dataframe.sampleBy-function:

`ValueError: 'map_from_arrays(array(0, 1), array(0.5, 0.25))' is not in list`